composer/algorithms/low_precision_groupnorm/low_precision_groupnorm.py

```diff
@@ -0,0 +1,99 @@
+
+
+class LPGroupNorm(torch.nn.GroupNorm):
+
+    def __init__(self, layer):
+        super().__init__(
+            num_groups=layer.num_groups,
+            num_channels=layer.num_channels,
+            eps=layer.eps,
+            affine=layer.affine,
+        )
+
+        with torch.no_grad():
+            self.weight.copy_(layer.weight)
+            self.bias.copy_(layer.bias)
+
+    def forward(self, x):
+        module_device = x.device
+        downcast_x = _cast_if_autocast_enabled(x)
+        downcast_weight = _cast_if_autocast_enabled(self.weight)
+        downcast_bias = _cast_if_autocast_enabled(self.bias)
+        with torch.autocast(enabled=False, device_type=module_device.type):
+            return F.group_norm(downcast_x, self.num_groups, downcast_weight, downcast_bias, self.eps)
+
+
+def to_LPGroupNorm(layer: torch.nn.Module, module_index: int) -> LPGroupNorm:
+    assert isinstance(layer,
+                      torch.nn.GroupNorm), 'The replacement policy will look for all instances of torch.nn.GroupNorm'
+    return LPGroupNorm(layer)
+
+
+def apply_low_precision_groupnorm(model, optimizers: Union[torch.optim.Optimizer, Sequence[torch.optim.Optimizer]],
+                                  precision: Precision):
+
+    if (precision != Precision.AMP_FP16 and precision != Precision.AMP_BF16):
+        warnings.warn(NoEffectWarning('Low Precision GroupNorm only applies to AMP_FP16 and AMP_BF16 precisions.'))
+        return model
+
+    policy: Dict[Type[torch.nn.Module], module_surgery.ReplacementFunction] = {torch.nn.GroupNorm: to_LPGroupNorm}
+
+    replaced_instances = module_surgery.replace_module_classes(module=model, optimizers=optimizers, policies=policy)
+    print(f'Replaced {len(replaced_instances)} instances of GroupNorm with LowPrecisionGroupNorm')
+    if len(replaced_instances) == 0:
+        warnings.warn(NoEffectWarning('No instances of torch.nn.GroupNorm found.'))
+    log.info(f'Successfully replaced {len(replaced_instances)} instances of GroupNorm with LowPrecisionGroupNorm')
+
+
+class LowPrecisionGroupNorm(Algorithm):
+    """
+    Replaces all instances of `torch.nn.GroupNorm` with `composer.algorithms.low_precision_groupnorm.low_precision_groupnorm.LPGroupNorm`.
+
+    LPGroupNorm is a thin wrapper around `torch.nn.GroupNorm` which forces the layer to run in lower precision (torch.float16 or torch.bfloat16)
+    if autocast is enabled. This algorithm has no effect in FP32 or DeepSpeed FP16 mode, where autocast is disabled.
+
+    This algorithm is intended to be used instead of Fused GroupNorm. They have similar behavior and performance.
+
+    Args:
+        apply_at (Event, optional): Event where algorithm is applied.
+    """
+
+    def __init__(self, apply_at: Optional[Event] = None):
+        self.apply_at = Event.INIT if apply_at is None else apply_at
+        if self.apply_at not in {Event.INIT, Event.AFTER_LOAD}:
```

Comment:
Is one of these two options a safe default? Would be nice to get have this algorithm work out of the box, but I know there are some thorny design issues here so no worries if that's infeasible.